modules/vba_documents/app/workers/vba_documents/report_monthly_submissions.rb

```diff
@@ -1,166 +1,49 @@
-
-    MEGABYTES = 1024.0 * 1024.0
-    def bytes_to_megabytes(bytes)
-      (bytes / MEGABYTES).round(2) unless bytes.nil? || bytes.zero?
-    end
-
-    def add_max_avg_mb
-      @monthly_max_avg.map! do |e|
-        h = { 'max_mb' => nil, 'avg_mb' => nil }
-        h['max_mb'] = bytes_to_megabytes(e['max_size'])
-        h['avg_mb'] = bytes_to_megabytes(e['avg_size'])
-        e.merge!(h)
-      end
-    end
-
-    def join_monthly_results
-      ret = []
-      @monthly_grouping.each_with_index do |base_row, idx|
-        base_row.merge!(@monthly_max_avg[idx]) if @monthly_max_avg[idx]
-        base_row.merge!(@monthly_mode[idx]) if @monthly_mode[idx]
-        ret << base_row
-      end
-      ret
-    end
+      prior_twelve_months_stats = []
 
-    def summary
-      sum_hash = { 'errored' => 0, 'expired' => 0, 'processing' => 0, 'success' => 0, 'vbms' => 0, 'total' => 0 }
-      @monthly_counts.each do |row|
-        sum_hash['errored'] += row['errored']
-        sum_hash['expired'] += row['expired']
-        sum_hash['processing'] += row['processing']
-        sum_hash['success'] += row['success']
-        sum_hash['vbms'] += row['vbms']
-        sum_hash['total'] += (row['errored'] + row['expired'] + row['processing'] + row['success'] + row['vbms'])
+      12.times do |i|
+        reporting_date = (i + 1).months.ago
+        prior_twelve_months_stats << stored_month_stats(reporting_date.month, reporting_date.year)
       end
-      sum_hash
-    end
 
-    # add the median pages to the modes hash
-    def get_median_results
-      @monthly_mode.each do |row|
-        zero_pad = "0#{row['mm']}"[-2..2]
-        yyyymm = "#{row['yyyy']}#{zero_pad}"
-        median = run_sql(SQLSupport::MEDIAN_SQL, yyyymm)
-        row['median_pages'] = median.first['median_pages']
-        row['mode_size'] = bytes_to_megabytes(row['mode_size'])
-        row['median_size'] = bytes_to_megabytes(median.first['median_size'])
-      end
+      VBADocuments::MonthlyReportMailer.build(prior_twelve_months_stats).deliver_now
     end
 
-    def rolling_status_times
-      results_set = []
-      12.times do |ym|
-        from = (ym + 1).months.ago.beginning_of_month
-        to = (ym + 1).month.ago.end_of_month
-        yyyymm = from.year.to_s + "00#{from.month}".last(2)
-        results = UploadSubmission.status_elapsed_times(from, to)
-        break if results.empty?
-
-        status_hash = reformat_rolling_times(results)
-        month_data = { yyyymm => status_hash }
-        get_elapsed_median(month_data)
-        tbs = get_time_between_statuses(yyyymm)
-        month_data[yyyymm] = month_data[yyyymm].merge!(tbs[yyyymm])
-        fmt_keys = %w[min_secs max_secs avg_secs median_secs min_bt_status max_bt_status avg_bt_status]
-        calc_times(month_data, fmt_keys)
-        results_set << month_data
-      end
-      results_set
+    def retry_limits_for_notification
+      # Notify at 1 day, 3 days
+      [14, 17]
     end
```

Comment:
Since we didn't have the retry notifications set up, this seemed like a good thing to add. In fact, we could use retry monitoring in most of our `VBADocuments` (and `VAForms`) jobs. I believe this is the first job where we've added it outside `AppealsApi`.